pygments/lexers/func.py

```diff
@@ -0,0 +1,67 @@
+from pygments.lexer import RegexLexer, include
+from pygments.token import Text, Comment, Operator, Keyword, Name, String, \
+    Number, Whitespace
+
+__all__ = ['FuncLexer']
+
+
+class FuncLexer(RegexLexer):
+    name = 'FunC'
+    aliases = ['func', 'fc']
+    filenames = ['*.fc', '*.func']
+
+    tokens = {
+        'root': [
+            (r'\n', Whitespace),
+            (r'\s+', Whitespace),
+
+            include('keywords'),
+            include('strings'),
+            include('directives'),
+            include('numeric'),
+            include('comments'),
+            include('storage'),
+            include('functions'),
+            include('variables'),
+        ],
+        'keywords': [
+            (r'(?<=\s)(<=>|>=|<=|!=|==|\^>>|\~>>|>>|<<|\/%|\^%|\~%|\^\/|\~\/|\+=|-=|\*=|\/=|~\/=|\^\/=|%=|\^%=|<<=|>>=|~>>=|\^>>=|&=|\|=|\^=|\^|=|~|\/|%|-|\*|\+|>|<|&|\||:|\?)(?=\s)', Operator),
+            (r'\b(if|ifnot|else|elseif|elseifnot|while|do|until|repeat|return|impure|method_id|forall|asm|inline|inline_ref)\b', Keyword),
+            (r'\b(false|true)\b', Keyword.Constant),
+        ],
+        'directives': [
+            (r'#include|#pragma', Keyword, 'directive'),
+        ],
+        'directive': [
+            include('strings'),
+            (r'version|not-version', Keyword),
+            (r'(>=|<=|=|>|<|\^)?([0-9]+)(.[0-9]+)?(.[0-9]+)?', Number), # version
+            (r';', Text, '#pop')
+        ],
+        'strings': [
+            (r'\"([^\n\"]+)\"(H|h|c|u|s|a)?', String),
+        ],
+        'numeric': [
+            (r'(-?([\d]+|0x[\da-fA-F]+))\b', Number)
```

Comment:
This regex won't cover cases like `0x111`
See https://pythex.org/?regex=(-%3F(%5B%5Cd%5D%2B%7C0x%5B%5Cda-fA-F%5D%2B))&test_string=0x123d&ignorecase=0&multiline=0&dotall=0&verbose=0

This also does not cover `b` numbers.
And numbers can have `_` in them.

It is better to use

```python
            (r'(0[xX][_\da-fA-F]+)', Number),
            (r'(0[bB][_0-1]+)', Number),
            (r'(-?\d[_\d]*)', Number),
```

Reply:
In this test you forgot to include boundary (\b)

https://pythex.org/?regex=(-%3F(%5B%5Cd%5D%2B%7C0x%5B%5Cda-fA-F%5D%2B))&test_string=0x123d&ignorecase=0&multiline=0&dotall=0&verbose=0

works nice